__tests__/rule-tester.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @fileoverview Helpers for tests.
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+'use strict'
+const { version } = require('eslint/package.json')
+const { RuleTester } = require('eslint')
+const globals = require('globals')
+
+const majorVersion = Number.parseInt(version.split('.')[0], 10)
+
+function convertConfig(config) {
+  if (config instanceof Object === false) {
+    return config
+  }
+
+  if (config.languageOptions == null) {
+    config.languageOptions = {}
+  }
+
+  if (config.parserOptions) {
+    Object.assign(config.languageOptions, config.parserOptions)
```

Comment:
please note `parserOptions` is not 100% same as `languageOptions`:

* https://eslint.org/docs/v8.x/use/configure/language-options#specifying-parser-options
* https://eslint.org/docs/latest/use/configure/language-options#specifying-javascript-options

Since this is only used for this project, it is not required to cover 100% of cases. so, it's be fine if you were not using other options. :)

Reply:
🤔 I was actually thinking about this, and there are a couple of things I could change:
1. I am considering making a module for this rule-tester compat package of some sort so it can be used in more places
2. It may be best to migrate the config backwards, not forwards. The more I think about it the less I like the `v8` -> `v9` compat, and instead prefer the idea of `v9` -> `v8`

Reply:
I also see the value for rule authors want to support eslint v8 & eslint v9. maybe support both - auto-detecting eslint version and the config format:

* eslint v8 + eslintrc : no change
* eslint v8 + flat : flat => eslintrc
* eslint v9 + eslintrc: eslintrc -> flat
* eslint v9 + flat: no change